ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -133,7 +140,7 @@ private List<Partition> getPartitions(Table table, AlterTableCompactDesc desc, D
       }
     } else {
       Map<String, String> partitionSpec = desc.getPartitionSpec();
-      partitions = context.getDb().getPartitions(table, partitionSpec);
+      partitions.addAll(PartitionUtils.getPartitions(context.getDb(), table, partitionSpec, false));
```

Comment:
why don't we use `partitionUtils.getPartition()` - it should return just 1 partition

Reply:
This is existing functionality which I didn't change.
I think it calls getPartitions on purpose to validate that only one partition is retrieved with the provided spec.
On the next lines it checks that only 1 partition was found for the partition spec and throws an exception otherwise.
```
      if (partitions.size() > 1) {
        throw new HiveException(ErrorMsg.TOO_MANY_COMPACTION_PARTITIONS);
      } else if (partitions.isEmpty()) {
        throw new HiveException(ErrorMsg.INVALID_PARTITION_SPEC);
      }
```

Reply:
Yes, this is used for scenarios like you said.
It is already supported for Iceberg with the change I did yesterday to PartitionUtils.getPartitions() method.
I verified it with a q-test and it throws the exception `TOO_MANY_COMPACTION_PARTITIONS` as expected.